### ISSUE TITLE
SDPA StableHLO export: explicit mask is broadcast invalidly under GQA and onto a dynamic key length

### ISSUE THREAD

**Issue by michalharakal** [SKaiNET-developers/SKaiNET#1302]
## Summary

`AttentionOperationsConverter` emits invalid StableHLO for an explicit attention mask in two cases:

1. **Per-head mask under grouped-query attention.** For a rank-4 mask `[b, H, Sq, Sk]` with `H` query heads and `nKV < H` K/V heads, the converter emits `broadcast_in_dim %mask, dims = [0, 1, 3, 4]` onto the grouped scores `[b, nKV, nRep, Sq, Sk]`. That maps the `H` axis onto the `nKV` axis, which is invalid even with fully static shapes. The head axis has to be *split* into `[nKV, nRep]`, not broadcast.
2. **Any mask whose key length is dynamic.** When the scores shape has a dynamic key length (the `?` cache dim of a KV-cache prefill/chunk graph) and the mask shape differs from the scores shape, the converter emits a static `stablehlo.broadcast_in_dim` whose result type is dynamic. StableHLO rejects that on every backend. This is not specific to grouped-query attention: a head-shared `[b, 1, Sq, ?]` mask onto multi-head scores `[b, H, Sq, ?]` hits it too.

The softmax in the same converter already handles the dynamic case with `dynamic_broadcast_in_dim`; the mask path was never given the same treatment.

## Reproduction

`SdpaGqaHloExportTest`-style graphs, current `develop` (e7c54b29):

| q | k/v | mask | emitted |
|---|---|---|---|
| `[1,4,8,16]` | `[1,2,8,16]` | `[1,4,8,8]` | `broadcast_in_dim %arg3, dims = [0, 1, 3, 4] : (tensor<1x4x8x8xf32>) -> tensor<1x2x2x8x8xf32>` |
| `[1,16,32,128]` | `[1,8,?,128]` | `[1,1,32,?]` | `broadcast_in_dim %arg3, dims = [0, 1, 3, 4] : (tensor<1x1x32x?xf32>) -> tensor<1x8x2x32x?xf32>` |
| `[1,16,32,128]` | `[1,8,?,128]` | `[1,16,32,?]` | `broadcast_in_dim %arg3, dims = [0, 1, 3, 4] : (tensor<1x16x32x?xf32>) -> tensor<1x8x2x32x?xf32>` |
| `[1,4,8,16]` | `[1,4,?,16]` | `[1,1,8,?]` | `broadcast_in_dim %arg3, dims = [0, 1, 2, 3] : (tensor<1x1x8x?xf32>) -> tensor<1x4x8x?xf32>` |

All four are rejected by `iree-compile` 3.11.0 on both `llvm-cpu` and `vulkan-spirv`, e.g.:

```
error: 'stablehlo.broadcast_in_dim' op result #0 must be statically shaped or single bounded dimension tensor of ...
```

Found while exporting the chunked-prefill graph (`prefill_with_past`, chunk 32, dynamic past length) of a grouped-query model (Qwen3-0.6B, 16 query heads, 8 K/V heads) from SKaiNET-transformers: the graph compiled on no backend.

## Expected

- Per-head mask under GQA: viewed as `[b, nKV, nRep, Sq, Sk]` with a reshape, in the same `h = kv * nRep + r` order the converter uses for Q; `dynamic_reshape` when a dim is dynamic.
- Mask onto a dynamic scores shape: `stablehlo.dynamic_broadcast_in_dim` with the scores-shape operand the softmax already builds.

## Backend note

IREE 3.11 only lowers `dynamic_broadcast_in_dim` for the mask when it carries `known_expanding_dimensions` / `known_nonexpanding_dimensions`: without them a dynamic operand dim mapped onto a dynamic result dim is ambiguous (1 to N or N to N) and legalization fails. IREE 3.11 does not lower `stablehlo.dynamic_reshape` at all, so a per-head mask with a dynamic key length is valid StableHLO after the fix but still not compilable by IREE 3.11; a head-shared `[b, 1, Sq, ?]` mask is the portable choice there.
